src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -1007,6 +1007,24 @@ class spell_dk_t20_2p_rune_empowered : public AuraScript
     int32 _runicPowerSpent = 0;
 };
 
+// 376905 - Unleashed Frenzy
+class spell_dk_unleashed_frenzy : public AuraScript
+{
+    bool CheckProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        if (Spell const* procSpell = eventInfo.GetProcSpell())
+            if (Optional<int32> cost = procSpell->GetPowerTypeCostAmount(POWER_RUNIC_POWER))
```

Comment:
This is wrong, this procs even if cast without cost (needs to check SpellInfo, not Spell) - triggers from Glacial Advance triggered from Arctic Assault